fix(php): improve the compatibility page (#15285)
newrelic/docs-website: src/content/docs/apm/agents/php-agent/getting-started/php-agent-compatibility-requirements.mdx

  
    To disable auto-update, you have two options: 
      * Add the PHP agent packages to an exclusion list for package upgrades, or
      * Version pin the PHP agent package to an agent version that supports the old, unsupported feature(s). The affected PHP agent packages are:
        * `newrelic-php5`
        * `newrelic-php5-common`
        * `newrelic-daemon`

## Permissions [#permissions]

Installing the PHP agent **requires** root access to:

* Read/write access to the PHP extensions directory
* Read/write access to PHP configuration data, either in the PHP configuration directory (preferred) or in `php.ini`.
* Ability to restart the web server or PHP process pool.

Running the agent **does not require** root access. The agent does need read/write access to the following files (the location of these files can be [customized in the PHP agent settings](/docs/php/php-agent-phpini-settings)):

> * [A file storing the agent log files](/docs/php/php-agent-phpini-settings#inivar-daemon-logfile)
> * [A file storing the agent socket endpoint](/docs/php/php-agent-phpini-settings#inivar-daemon-port)
> * [A file storing the agent process ID](/docs/php/php-agent-phpini-settings#inivar-daemon-pidfile)
> 

Author: ```suggestion
* [A file storing the agent log files](/docs/apm/agents/php-agent/configuration/php-agent-configuration/#inivar-daemon-logfile)
* [A file storing the agent socket endpoint](/docs/apm/agents/php-agent/configuration/php-agent-configuration/#inivar-daemon-port)
* [A file storing the agent process ID](/docs/apm/agents/php-agent/configuration/php-agent-configuration/#inivar-daemon-pidfile)

```

  text: Monitor and improve your entire stack. 100GB free. Forever.
---

Before you [install the PHP agent](/docs/apm/agents/php-agent/installation), make sure your system meets the version requirements listed below.

## PHP releases [#php-release]

The following PHP versions are supported:

<table>
    <thead>
        <tr>
            <th style={{ width: "250px" }}>
                Version
            </th>
            <th>
                Compatibility notes
            </th>
        </tr>
    </thead>
    <tbody>

>         <tr>
>             <td>
>                 7.0 - 7.4
>             </td>
>         </tr>

Author: ```suggestion
        <tr>
            <td>
                7.0 - 7.4
            </td>
                       <td>
                -
            </td>
        </tr>
```

  text: Monitor and improve your entire stack. 100GB free. Forever.
---

Before you [install the PHP agent](/docs/apm/agents/php-agent/installation), make sure your system meets the version requirements listed below.

## PHP releases [#php-release]

The following PHP versions are supported:

<table>
    <thead>
        <tr>
            <th>
                Version
            </th>
            <th>
                Compatibility notes
            </th>
        </tr>
    </thead>
    <tbody>
        <tr>
            <td>
                7.0 - 7.4
            </td>

>             <td>
>                 7.0 and 7.1 are for EOL in December 2023.
>             </td>

Reviewer: ```suggestion
```

        </tr>
        <tr>
            <td>
                8.0 - 8.2
            </td>
            <td>
                When PHP 8.0, 8.1, or 8.2 detects an observability extension (the PHP agent),  [Just-In-Time compilation](https://php.watch/articles/jit-in-depth) is disabled.

                PHP 8.1 and later does not support [Fibers](https://www.php.net/manual/en/language.fibers.php).
            </td>
        </tr>
    </tbody>
</table> 

## PHP agent compatibility

See [PHP agent release notes](/docs/release-notes/agent-release-notes/php-release-notes/) for the latest PHP agent release.

>   * For PHP builds that are compiled with Zend Thread Safety (ZTS), only [PHP agent 9.19](/docs/release-notes/agent-release-notes/php-release-notes/php-agent-9190309) and later supports these builds.

Reviewer: ```suggestion
  * For PHP builds that are compiled with Zend Thread Safety (ZTS), only [PHP agent 9.19](/docs/release-notes/agent-release-notes/php-release-notes/php-agent-9190309) and earlier supports these builds.
```

  </tbody>
</table>

## Other third-party monitoring software [#other-monitoring-software]

If your application uses any additional monitoring software, we cannot guarantee that our agent will work correctly and cannot offer technical support. For more information, see [Errors when using other monitoring software](/docs/apm/new-relic-apm/troubleshooting/errors-while-using-new-relic-apm-alongside-other-apm-software).

> ## Instance details

Reviewer 2: @ally-sassman, this looks like it's related to the main database section above. You might check with the SME to see if there's a way to combine these database instance details with the other database section. Or, consider at least moving this right below the database section.

      <td>

>         [redis](https://pecl.php.net/package/redis)
>         [predis](https://github.com/nrk/predis)

Author: ```suggestion
        * [redis](https://pecl.php.net/package/redis)
        * [predis](https://github.com/nrk/predis)
```

    <tr>
      <td>
        [Codestream (code-level metrics)](/docs/codestream/how-use-codestream/performance-monitoring/)
      </td>

      <td>
        New Relic CodeStream brings observability into the IDE to show how your code is performing in production at the method level.

>         The PHP agent versions 7.0 - 7.4 and 8.0 - 8.2 report code-level metrics (starting with release 10.4.0).

Reviewer: ```suggestion
        PHP versions 7.0 - 7.4 and 8.0 - 8.2 report code-level metrics (starting with release 10.4.0).
```
It is versions of PHP, not the agent. Though I am unsure how necessary this list is, as it encompasses (and will continue to encompass in the future) all PHP versions supported by the agent.
